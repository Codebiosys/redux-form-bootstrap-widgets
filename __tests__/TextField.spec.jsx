/* eslint-disable import/first */

import React from 'react';
import { mount } from 'enzyme';
import { InputGroup } from 'react-bootstrap';

import { TextField } from 'index';

const fieldProps = {
  input: {
    name: 'fieldName',
    onChange: jest.fn(),
  },
  meta: {
    form: 'formName',
    pristine: false,
  },
  label: 'Text Test',
  helpText: 'The help Text',
  addOnBefore: (<div id="before_addon">before</div>),
  addOnAfter: (<div id="after_addon">after</div>),
};

const valueFieldProps = { ...fieldProps };
valueFieldProps.input = { ...fieldProps.input };
valueFieldProps.input.value = 'entered text';
valueFieldProps.input.value = 'entered text';
const textAreaFieldProps = { ...fieldProps };
textAreaFieldProps.type = 'textarea';


describe('The Text Field', () => {
  let inputWrapper;
  let textAreaWrapper;
  let valueWrapper;
  beforeEach(() => {
    inputWrapper = mount(<TextField {...fieldProps} />);
    valueWrapper = mount(<TextField {...valueFieldProps} />);
    textAreaWrapper = mount(<TextField {...textAreaFieldProps} />);
  });

  it('renders', () => {
    expect(inputWrapper).toMatchSnapshot();
  });

  it('displays the add ons and clear button for text input', () => {
    expect(inputWrapper.find('#before_addon').exists()).toBe(true);
    expect(inputWrapper.find('#after_addon').exists()).toBe(true);
    expect(inputWrapper.find('button').exists()).toBe(true);
  });

  it('disables clear button when there is no value', () => {
    expect(inputWrapper.find('button').prop('disabled')).toEqual(true);
  });

  it('enables clear button when there is a value', () => {
    expect(valueWrapper.find('button').prop('disabled')).toEqual(false);
  });

  it('clears the value when the clear button is pressed', () => {
    valueWrapper.find('button').simulate('click');
    expect(valueFieldProps.input.onChange).toHaveBeenCalledWith(null);
  });

  it('displays a textarea when the type is textarea', () => {
    expect(textAreaWrapper).toMatchSnapshot();
    expect(textAreaWrapper.find('textarea').exists()).toBe(true);
  });

  it('displays full width when the type is textarea', () => {
    expect(textAreaWrapper.find(InputGroup).prop('style')).toEqual({ width: '100%' });
  });

  it('hides the add ons and clear button for textareas', () => {
    expect(textAreaWrapper.find('#before_addon').exists()).toBe(false);
    expect(textAreaWrapper.find('#after_addon').exists()).toBe(false);
    expect(textAreaWrapper.find('button').exists()).toBe(false);
  });
  it('has the password field type when set to password', () => {
    const passwordFieldProps = { ...fieldProps };
    passwordFieldProps.type = 'password';
    const passwordWrapper = mount(<TextField {...passwordFieldProps} />);
    expect(passwordWrapper.find('input').prop('type')).toEqual('password');
  });
});
