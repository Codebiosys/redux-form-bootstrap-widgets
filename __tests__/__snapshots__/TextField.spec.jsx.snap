// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text Field displays a textarea when the type is textarea 1`] = `
<TextField
  addOnAfter={
    <div
      id="after_addon"
    >
      after
    </div>
  }
  addOnBefore={
    <div
      id="before_addon"
    >
      before
    </div>
  }
  customValidation={null}
  disabled={false}
  helpText="The help Text"
  input={
    Object {
      "name": "fieldName",
      "onChange": [Function],
    }
  }
  label="Text Test"
  meta={
    Object {
      "form": "formName",
      "pristine": false,
    }
  }
  required={false}
  type="textarea"
>
  <FormGroup
    bsClass="form-group"
    controlId="fieldName"
  >
    <div
      className="form-group"
    >
      <Label
        label="Text Test"
        required={false}
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          <label
            className="control-label"
            htmlFor="fieldName"
          >
            Text Test
          </label>
        </ControlLabel>
      </Label>
      <InputGroup
        bsClass="input-group"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <span
          className="input-group"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <FormControl
            bsClass="form-control"
            componentClass="textarea"
            disabled={false}
            name="fieldName"
            onChange={[Function]}
            style={
              Object {
                "zIndex": "0",
              }
            }
          >
            <textarea
              className="form-control"
              disabled={false}
              id="fieldName"
              name="fieldName"
              onChange={[Function]}
              style={
                Object {
                  "zIndex": "0",
                }
              }
            />
          </FormControl>
        </span>
      </InputGroup>
      <HelpBlock
        bsClass="help-block"
        style={
          Object {
            "minHeight": "6ex",
          }
        }
      >
        <span
          className="help-block"
          style={
            Object {
              "minHeight": "6ex",
            }
          }
        >
          The help Text
        </span>
      </HelpBlock>
    </div>
  </FormGroup>
</TextField>
`;

exports[`The Text Field renders 1`] = `
<TextField
  addOnAfter={
    <div
      id="after_addon"
    >
      after
    </div>
  }
  addOnBefore={
    <div
      id="before_addon"
    >
      before
    </div>
  }
  customValidation={null}
  disabled={false}
  helpText="The help Text"
  input={
    Object {
      "name": "fieldName",
      "onChange": [Function],
    }
  }
  label="Text Test"
  meta={
    Object {
      "form": "formName",
      "pristine": false,
    }
  }
  required={false}
  type="text"
>
  <FormGroup
    bsClass="form-group"
    controlId="fieldName"
  >
    <div
      className="form-group"
    >
      <Label
        label="Text Test"
        required={false}
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          <label
            className="control-label"
            htmlFor="fieldName"
          >
            Text Test
          </label>
        </ControlLabel>
      </Label>
      <InputGroup
        bsClass="input-group"
        style={Object {}}
      >
        <span
          className="input-group"
          style={Object {}}
        >
          <div
            id="before_addon"
          >
            before
          </div>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            disabled={false}
            name="fieldName"
            onChange={[Function]}
            style={
              Object {
                "zIndex": "0",
              }
            }
            type="text"
          >
            <input
              className="form-control"
              disabled={false}
              id="fieldName"
              name="fieldName"
              onChange={[Function]}
              style={
                Object {
                  "zIndex": "0",
                }
              }
              type="text"
            />
          </FormControl>
          <div
            id="after_addon"
          >
            after
          </div>
        </span>
      </InputGroup>
      <HelpBlock
        bsClass="help-block"
        style={
          Object {
            "minHeight": "6ex",
          }
        }
      >
        <span
          className="help-block"
          style={
            Object {
              "minHeight": "6ex",
            }
          }
        >
          The help Text
        </span>
      </HelpBlock>
    </div>
  </FormGroup>
</TextField>
`;
